### PR TITLE
Reference N817 from N814, and vice-versa

### PR DESCRIPTION
**Pull request by jparise**
Also add anchors and pep8.org references for many of the rules.

See #165

### CODE REVIEW

**Comment by sigmavirus24** [comment]
I don't believe this explains the separate error codes which is what was requested in there by Sam

**Comment by jparise** [comment]
Yes, I totally agree. I was thinking this could be a useful small improvement.

But if a larger documentation section explaining the codes is in the works, then that's even better.

File: README.rst [hunk 1]
```diff
@@ -59,13 +59,13 @@ These error codes are emitted:
...
 +------+-------------------------------------------------------+
-| N814 | camelcase imported as constant                        |
+| N814 | camelcase imported as constant (see also ``N817``)    |
 +------+-------------------------------------------------------+
 | N815 | mixedCase variable in class scope                     |
 +------+-------------------------------------------------------+
 | N816 | mixedCase variable in global scope                    |
 +------+-------------------------------------------------------+
-| N817 | camelcase imported as acronym                         |
+| N817 | camelcase imported as acronym (see also ``N814``)     |
```
Comment (sigmavirus24) [inline README.rst:68]:
```suggestion
| N817 | camelcase imported as acronym (see also ``N814``) |
|      | separated from ``N814`` for easy ignoring         |
```

I'm pretty sure that rST allows you to have multiple lines in a table row like this. I think this is all that would satisfy the request in the original issue.

**Comment by jparise** [comment]
I took another pass on this section, adding pep8.org references where they were obvious, as well.

File: README.rst [hunk 2]
```diff
@@ -35,38 +35,48 @@ Error Codes
...
-+------+-------------------------------------------------------+
-| N816 | mixedCase variable in global scope                    |
-+------+-------------------------------------------------------+
-| N817 | camelcase imported as acronym                         |
-+------+-------------------------------------------------------+
++---------+---------------------------------------------------------+
+| code    | sample message                                          |
++=========+=========================================================+
+| _`N801` | class names should use CapWords convention              |
+|         | (https://pep8.org/#class-names)                         |
```
Comment (sigmavirus24) [inline README.rst:42]: Please use the real pep link. I don't want to drive traffic to someone who raised over 30k for a project and then ran away with the funds.

Comment (jparise) [reply README.rst:42]: I'll change them. I started with python.org URLs, but they're much longer, and reST makes you inflate the table to the maximum cell's width. Maybe I'll avoid that using hyperlink aliases or something.

Comment (sigmavirus24) [reply README.rst:42]: You can do things like:

```rst
`class names should use CapWords convention <class-names>`_
```

And later in the file do


```rst
.. _class-names: https://python.org...
```

File: README.rst [hunk 3]
```diff
@@ -35,40 +35,51 @@ Error Codes
...
-| N816 | mixedCase variable in global scope                    |
-+------+-------------------------------------------------------+
-| N817 | camelcase imported as acronym                         |
-+------+-------------------------------------------------------+
-| N818 | error suffix in exception names                       |
-+------+-------------------------------------------------------+
++---------+-----------------------------------------------------------------+
+| code    | sample message                                                  |
++=========+=================================================================+
+| _`N801` | class names should use CapWords convention (`<class-names>`_)   |
```
Comment (sigmavirus24) [inline README.rst:41]:
```suggestion
| _`N801` | class names should use CapWords convention (`in PEP 8 <class-names>`_)   |
```

So I don't think an anchor like that _just works_ and you'll need some text to display. See the suggestion above

File: README.rst [hunk 4]
```diff
@@ -35,40 +35,51 @@ Error Codes
...
++---------+-----------------------------------------------------------------+
+| _`N818` | error suffix in exception names (`<exceptions>`_)               |
++---------+-----------------------------------------------------------------+
+
+.. class-names: https://www.python.org/dev/peps/pep-0008/#class-names
+.. constants: https://www.python.org/dev/peps/pep-0008/#constants
+.. exceptions: https://www.python.org/dev/peps/pep-0008/#exception-names
+.. function-names: https://www.python.org/dev/peps/pep-0008/#function-and-variable-names
+.. function-args: https://www.python.org/dev/peps/pep-0008/#function-and-method-arguments
+.. method-names: https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables
```
Comment (sigmavirus24) [inline README.rst:82]:
```suggestion
.. _class-names: https://www.python.org/dev/peps/pep-0008/#class-names
.. _constants: https://www.python.org/dev/peps/pep-0008/#constants
.. _exceptions: https://www.python.org/dev/peps/pep-0008/#exception-names
.. _function-names: https://www.python.org/dev/peps/pep-0008/#function-and-variable-names
.. _function-args: https://www.python.org/dev/peps/pep-0008/#function-and-method-arguments
.. _method-names: https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables
```

File: README.rst (hunk 3 above)
Comment (jparise) [reply README.rst:41]: I was working, but I like your suggestion better.

**Review by sigmavirus24 — APPROVED** [review]

**Pull request merged**